### PR TITLE
New forum setup updates batch #3

### PR DESCRIPTION
**Pull request by Stvad**
Another batch of updates/fixes from setting up https://modernswing.forum 
As with previous batches - I recommend looking at commits one by one

- **Make EALoginPopover.tsx work better for all friendlyUI sites #to-upstream**
- **Move initializeSetting into settingsCache.ts so db settings can have a dependency on instance settings without a cycle #to-upstream**
- **Introduce showSubscribeReminderInFeed setting, to allow forums to choose to hide a reminder to subscribe from main page feed #to-upstream**
- **Make useShortAllTagsPath default for Friendly UI #to-upstream**
- **Auth0 comment update about grant types #to-upstream**
- **A better default for homepagePostFeedsSetting, so LWHome is not broken by default #to-upstream**
- **Make notifications page available in friendly ui mode (default) #to-upstream**
- **Simplify quicktakes route bc we want the same behaviour everywhere #to-upstream**
- **Create assumeUserEmailVerifiedSetting, instead of having it hardcoded that EA doesn't verify emails #to-upstream**
- **Use taggingNameSetting in EARecentDiscussionTagRevision instead of "tag" #to-upstream**
- **Name default link to shortform in left sidebar "Quick takes" to align with the rest of the defaults #to-upstream**
- **Make a quickTakesTagsEnabledSetting (vs being hardcoded for EA forum) #to-upstream**

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207184986824018) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (oetherington) [inline packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:348]: This should probably use the `siteNameWithArticle` setting

File: packages/lesswrong/lib/routes.ts
Comment (Stvad) [reply packages/lesswrong/lib/routes.ts:1306]: Is description functional? (that is do we display it somewhere?)

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:472]: whoops, should have highlighted that!

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:348]: that'd become `the Effective Altruism Forum` vs `the EA Forum` in that case though. (happy to change it, but was trying to keep existing functionality unchanged)

Comment (oetherington) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:348]: The `siteNameWithArticle` setting does actually (confusingly) use the short name (at least it does on the EA forum).

It's not a big deal - I don't really mind either way, but it might be weird in the future for any forums that don't want to add "the" to their name.

File: packages/lesswrong/lib/routes.ts
Comment (oetherington) [reply packages/lesswrong/lib/routes.ts:1306]: Oh - I thought we showed it in a tooltip in the menu, but I just checked again and it turns out we only show the tooltip for "large" menu items, not "small" ones. It's probably fine to remove it then.

Screenshot 2024-05-01 at 13 26 41

Screenshot 2024-05-01 at 13 26 44

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (agnesstenlund) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:472]: Oh, if we don't need to show it since you've already agreed it seems totally fine (better?) to remove it

File: packages/lesswrong/lib/routes.ts
Comment (Stvad) [reply packages/lesswrong/lib/routes.ts:1306]: I believe that label is coming from “tooltip” field in menuTabs and not from the route description

**Comment by s-cheng** [comment]
@Stvad Thanks so much for all the improvements here! Apologies, I've been a bit behind on things and am leaving for vacation now so one of the other devs will take care of the code review

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:475]: I figured I just remove a link here tbh, resulting in something like:
image

lmk if that works!

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx]: Updated it to use `siteNameWithArticle`

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx]: @oetherington  @agnesstenlund would appreciate your comments on this! 
@oetherington also lmk if you have any concerns about this PR!

File: packages/lesswrong/server/authentication/auth0.ts
Comment (Will-Howard) [inline packages/lesswrong/server/authentication/auth0.ts:188]: Nice

**Review by Will-Howard — APPROVED** [review]
LGTM! I'll just wait for @agnesstenlund to reply about the ea.org change before merging

File: packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:471]: Yeah, see this thread for more context: https://github.com/ForumMagnum/ForumMagnum/pull/9193#discussion_r1588542693

Comment (Will-Howard) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:471]: Ah sorry I didn't see that comment, anyway we think this is fine. Could you just fix that conflict, then I'll merge

Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:478]: Fixed the conflict :)

Comment (Stvad) [reply packages/lesswrong/components/ea-forum/auth/EALoginPopover.tsx:478]: (I figure I'll just merge before it develops more conflicts, given that you've approved this and we're ok with this display change)

**Pull request merged**
